GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -167,11 +171,28 @@ private void ForceProjectionChange(ITracer tracer, GVFSEnlistment enlistment)
             if (!this.ShowStatusWhileRunning(
                 () =>
                 {
-                    GitProcess git = new GitProcess(enlistment);
-                    GitProcess.Result checkoutResult = git.ForceCheckout("HEAD");
+                    NamedPipeMessages.PostIndexChanged.Response response = null;
 
-                    errorMessage = checkoutResult.Errors;
-                    return checkoutResult.ExitCodeIsSuccess;
+                    try
+                    {
+                        using (NamedPipeClient pipeClient = new NamedPipeClient(enlistment.NamedPipeName))
+                        {
+                            if (!pipeClient.Connect())
+                            {
+                                this.ReportErrorAndExit("Unable to connect to GVFS.  Try running 'gvfs mount'");
+                            }
+
+                            NamedPipeMessages.PostIndexChanged.Request request = new NamedPipeMessages.PostIndexChanged.Request(updatedWorkingDirectory: true, updatedSkipWorktreeBits: false);
+                            pipeClient.SendRequest(request.CreateMessage());

```

Comment:
Interesting that you run a post-index-changed message instead of `git checkout`, but that's a safe way to communicate with the Mount process without changing contents.